From 2d81043e7c98b31b37a1ecd1f037a04c60e662aa Mon Sep 17 00:00:00 2001
From: Roy Li <rongqing.li@windriver.com>
Date: Mon, 10 Feb 2014 18:10:12 +0800
Subject: [PATCH 31/34] policy/module/ftp: add ftpd_t to
 mls_file_write_all_levels

Proftpd will create file under /var/run, but its mls is in high, and
can not write to lowlevel

Upstream-Status: Pending

type=AVC msg=audit(1392347709.621:15): avc:  denied  { write } for  pid=545 comm="proftpd" name="/" dev="tmpfs" ino=5853 scontext=system_u:system_r:ftpd_t:s15:c0.c1023 tcontext=system_u:object_r:var_run_t:s0-s15:c0.c1023 tclass=dir
type=AVC msg=audit(1392347709.621:15): avc:  denied  { add_name } for  pid=545 comm="proftpd" name="proftpd.delay" scontext=system_u:system_r:ftpd_t:s15:c0.c1023 tcontext=system_u:object_r:var_run_t:s0-s15:c0.c1023 tclass=dir
type=SYSCALL msg=audit(1392347709.621:15): arch=c000003e syscall=2 success=yes exit=3 a0=471910 a1=42 a2=1b6 a3=8 items=0 ppid=539 pid=545 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 ses=4294967295 tty=(none) comm="proftpd" exe="/usr/sbin/proftpd" subj=system_u:system_r:ftpd_t:s15:c0.c1023 key=(null)

root@localhost:~# sesearch --allow -s ftpd_t -t var_run_t|grep dir|grep add_name
   allow ftpd_t var_run_t : dir { ioctl read write getattr lock add_name remove_name search open } ;
root@localhost:~#

Signed-off-by: Roy Li <rongqing.li@windriver.com>
Signed-off-by: Joe MacDonald <joe_macdonald@mentor.com>
---
 policy/modules/services/ftp.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/ftp.te b/policy/modules/services/ftp.te
index 29bc077c..d582cf80 100644
--- a/policy/modules/services/ftp.te
+++ b/policy/modules/services/ftp.te
@@ -150,6 +150,8 @@ role ftpdctl_roles types ftpdctl_t;
 type ftpdctl_tmp_t;
 files_tmp_file(ftpdctl_tmp_t)
 
+mls_file_write_all_levels(ftpd_t)
+
 type sftpd_t;
 domain_type(sftpd_t)
 role system_r types sftpd_t;
-- 
2.19.1

